sorting/merge_sort.cpp

```diff
@@ -54,8 +53,17 @@ void merge(int *arr, int l, int m, int r) {
         k++;
     }
 
-    delete[] L;
-    delete[] R;
+    while (i < n1) {
+        arr[k] = L[i];
```

Comment:
what is this doing?

Reply:
These loops are necessary because, in some cases, either L or R may still contain elements after the main merging loop has finished. These remaining elements are already sorted within their respective arrays, so they can be directly copied into arr.